### PR TITLE
Added Descriptive Labels to Branch List on Branches page and Tags List on Tags Page

### PR DESCRIPTION
**Pull request by lmelende20**
There are errors in the ComponentNodes.tsx file on the ariaLabel prop however, the file runs and performs as desired. I tested it with the VoiceOver screen reader, and all of the branches on the "Overview" and "All branches" pages were read as desired.

## App preview:

- [Web](https://sg-web-lm-branchlistlabels.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-rwduoeyqwr.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by lmelende20** [comment]
I would love some input and advice as to how to fix the errors in the ConnectionNodes.tsx file.

**Comment by lmelende20** [comment]
Regarding issues #35461, "Branches page: Undescriptive label on branch list", and #35467, "Tags page: Undescriptive label on tags list". This is an abstract approach to the issue, so as to not disrupt any other files that rely on FilteredConnection.tsx and ConnectionNodes.tsx. There were several issues with typing, so I followed patterns in FilteredConnection.tsx and Connection.tsx to resolve them the best I could. I'm not sure if there is a better approach, but I would love any feedback on my solution!

File: client/web/src/repo/GitReference.tsx [hunk 1]
```diff
@@ -39,6 +39,8 @@ export interface GitReferenceNodeProps {
 
     onClick?: React.MouseEventHandler<HTMLAnchorElement>
     nodeLinkClassName?: string
+
+    ariaLabel?: string
```
Comment (jasongornall) [inline client/web/src/repo/GitReference.tsx:43]: should this be required?

File: client/web/src/components/FilteredConnection/utils.ts [hunk 2]
```diff
@@ -13,6 +13,12 @@ import type { FilteredConnectionFilter, FilteredConnectionFilterValue } from './
 export const hasID = (value: unknown): value is { id: Scalars['ID'] } =>
     typeof value === 'object' && value !== null && hasProperty('id')(value) && typeof value.id === 'string'
 
+export const hasDisplayName = (value: unknown): value is { displayName: Scalars['String'] } =>
```
Comment (jasongornall) [inline client/web/src/components/FilteredConnection/utils.ts:16]: this file confuses me.. is the goal to mirror a kind of lodash.get?

File: client/web/src/components/FilteredConnection/FilteredConnection.tsx [hunk 3]
```diff
@@ -532,8 +536,10 @@ export class FilteredConnection<
                         emptyElement={this.props.emptyElement}
                         totalCountSummaryComponent={this.props.totalCountSummaryComponent}
                         withCenteredSummary={this.props.withCenteredSummary}
+                        ariaLabelFunction={this.props.ariaLabelFunction}
                     />
                 )}
+                {/* {console.log(this.props)} */}
```
Comment (jasongornall) [inline client/web/src/components/FilteredConnection/FilteredConnection.tsx:542]: ?

File: client/web/src/repo/GitReference.tsx (hunk 1 above)
Comment (lmelende20) [reply client/web/src/repo/GitReference.tsx:43]: Without that, I get an error "Property 'ariaLabel' does not exist on type 'PropsWithChildren<PropsWithChildren<GitReferenceNodeProps>>'."

File: client/web/src/components/FilteredConnection/FilteredConnection.tsx (hunk 3 above)
Comment (lmelende20) [reply client/web/src/components/FilteredConnection/FilteredConnection.tsx:542]: I left some of my error-checking in the file by accident. I will delete them and commit again

**Review by jasongornall — APPROVED** [review]

**Pull request merged**